chaos-monkey-spring-boot/src/main/java/de/codecentric/spring/boot/chaos/monkey/configuration/toggles/UnleashChaosToggles.java

```diff
@@ -0,0 +1,29 @@
+package de.codecentric.spring.boot.chaos.monkey.configuration.toggles;
+
+import de.codecentric.spring.boot.chaos.monkey.component.MetricType;
+import de.codecentric.spring.boot.chaos.monkey.configuration.ChaosMonkeyProperties;
+import no.finn.unleash.Unleash;
+
+public class UnleashChaosToggles implements ChaosToggles {
+  private final ChaosMonkeyProperties chaosMonkeyProperties;
+  private final Unleash unleash;
+
+  public UnleashChaosToggles(ChaosMonkeyProperties chaosMonkeyProperties, Unleash unleash) {
+    this.chaosMonkeyProperties = chaosMonkeyProperties;
+    this.unleash = unleash;
+  }
+
+  @Override
+  public String nameMapper(MetricType type, String name) {
+    if (type == null) {
+      return chaosMonkeyProperties.getTogglePrefix() + ".all";
+    }
+
+    return chaosMonkeyProperties.getTogglePrefix() + "." + type.getMetricName();
```

Comment:
Here you can see that with Unleash the toggles get grouped by the metric type. So by default, if Chaos Monkey if running and enabled the following toggles would be getting checked:

* chaos.monkey.all
* chaos.monkey.controller
* chaos.monkey.restController
* chaos.monkey.repository
* chaos.monkey.service
* chaos.monkey.component

Put with the name mapping a user could override it and divide it further based on the name input. They can change the prefix, since it is common to prefix toggle based on and application:

* myApp.chaos.monkey.repository
* myOtherApp.chaos.monkey.repository

Thus allowing easy segmentation of toggles between apps, if that is desired.